src/libcore/num/mod.rs

```diff
@@ -4580,6 +4602,29 @@ impl_from! { u32, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0"
 // Float -> Float
 impl_from! { f32, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0")] }
 
+// Integer -> Boolean
+impl_from! { u8, bool, #[unstable(feature = "into_bool", issue = "0")]}
+impl_from! { u16, bool, #[unstable(feature = "into_bool", issue = "0")]}
+impl_from! { u32, bool, #[unstable(feature = "into_bool", issue = "0")]}
+impl_from! { u64, bool, #[unstable(feature = "into_bool", issue = "0")]}
+impl_from! { u128, bool, #[unstable(feature = "into_bool", issue = "0")]}
+impl_from! { i8, bool, #[unstable(feature = "into_bool", issue = "0")]}
+impl_from! { i16, bool, #[unstable(feature = "into_bool", issue = "0")]}
+impl_from! { i32, bool, #[unstable(feature = "into_bool", issue = "0")]}
+impl_from! { i64, bool, #[unstable(feature = "into_bool", issue = "0")]}
+impl_from! { i128, bool, #[unstable(feature = "into_bool", issue = "0")]}
+// Boolean -> Integer
```

Comment:
There's a PR open for this part at https://github.com/rust-lang/rust/pull/50554

Reply:
I updated the issue number to #46109